workflow-testing/src/main/java/com/squareup/workflow1/testing/RenderTesterWorkers.kt

```diff
@@ -175,7 +175,16 @@ fun <PropsT, StateT, OutputT, RenderingT>
       output = output,
       exactMatch = true
   ) { actualWorkerType, worker, actualKey ->
-    (key == actualKey && workerType.isSupertypeOf(actualWorkerType))
+    val ruleExpectsNothing = typeOf<Worker<Nothing>>().isSupertypeOf(workerType)
+    val actualExpectsNothing = typeOf<Worker<Nothing>>().isSupertypeOf(actualWorkerType)
+
+    // We have to take some care when the actual worker is `Worker<Nothing>`, b/c
+    // Worker<Something>.isSupertypeOf(Worker<Nothing>) is always true -- Nothing is
+    // the bottom type. So, we only make that check if the rule is `Worker<Nothing>`,
+    // or both the rule and the actual are `Worker<Something>`.
+
+    (key == actualKey &&
+        (ruleExpectsNothing || !actualExpectsNothing) && workerType.isSupertypeOf(actualWorkerType))
```

Comment:
I am glad you added that comment and tests for this, i am having trouble figuring out what this actually means just from the code.